tests/cypress/scripts/cliHelper.js

```diff
@@ -68,94 +255,19 @@ export const cliHelper = {
-      cy.log(res.stdout ? res.stdout : res.stderr)
-    })
-  },
-  createApplication: (appName, namespace) => {
-    cy.readFile('tests/cypress/templates/application.yaml').then((cfg) => {
-      let b64Cfg = btoa(
-        cfg.replaceAll('APPNAME', appName).replaceAll('NAMESPACE', namespace)
-      )
-      cy.exec(`echo ${b64Cfg} | base64 -d | oc apply -f -`)
-      cy.log(`Successfully created application (${appName})`)
-    })
-  },
-  deleteNamespace: (name, kubeconfig = '') => {
-    cy.exec(`${kubeconfig} oc delete namespace ${name}`).then((res) => {
-      cy.log(res.stdout ? res.stdout : res.stderr)
-    })
-  },
-  login: (mode) => {
-    var mode = mode === 'Local' ? 'HUB' : 'MANAGED'
+
+  /**
+   * Login into the cluster environment with the `oc` cli command.
+   * @param {string} cluster The cluster environment to login into (Default: HUB).
+   * @param {object} options Additional options for logging into the cluster environment.
+   */
+  login: (cluster = 'HUB', options = {}) => {
     cy.exec(
       `oc login --server=https://api.${Cypress.env(
-        `OPTIONS_${mode}_BASEDOMAIN`
-      )}:6443 -u ${Cypress.env(`OPTIONS_${mode}_USER`)} -p ${Cypress.env(
-        `OPTIONS_${mode}_PASSWORD`
+        `OPTIONS_${cluster}_BASEDOMAIN`
+      )}:6443 -u ${Cypress.env(`OPTIONS_${cluster}_USER`)} -p ${Cypress.env(
+        `OPTIONS_${cluster}_PASSWORD`
       )} --insecure-skip-tls-verify`
     )
   },
-  setup: (modes) => {
-    modes.forEach((mode) => {
-      if (!mode.skip) {
-        describe(
-          `Search: Create resource in ${mode.label} Cluster`,
-          { tags: tags.env },
-          function () {
-            // Log into the hub and managed cluster with the oc command to create the resources.
-            context(
-              `prereq: create resource with oc command`,
-              { tags: tags.required },
-              function () {
-                it(`[P1][Sev1][${squad}] should log into ${mode.label.toLocaleLowerCase()} cluster`, function () {
-                  if (
-                    mode.label === 'Managed' &&
-                    Cypress.env('USE_MANAGED_KUBECONFIG')
-                  ) {
-                    cy.log('Skipping login and using kubeconfig file')
-                  } else {
-                    cliHelper.login(mode.label)
-                  }
-                })
-
-                it(`[P1][Sev1][${squad}] should create namespace resource`, function () {
-                  cliHelper.createNamespace(mode.namespace, mode.kubeconfig)
-                })
-
-                it(`[P1][Sev1][${squad}] should create deployment resource`, function () {
-                  cliHelper.createDeployment(
-                    mode.namespace + '-deployment',
-                    mode.namespace,
-                    'openshift/hello-openshift',
-                    mode.kubeconfig
-                  )
-                })
-              }
-            )
-          }
-        )
-      }
-    })
-  },
```

Comment:
The setup steps will now be contained within each test `beforeEach` hook. It will create the resource and set an environment variable state to `true` for that resource. This will allow the test to know whether the resource was created or not.